### PR TITLE
Build Widget Configuration

### PR DESCRIPTION
**Pull request by jhoelscher7**
- Routing to build widget based on dashboard id
- Cleaned up build widget and added titles
image
- Ability to configure and save build widget settings in modal form
image
- Typeahead in build widget configuration
- Widget refresh at fixed interval (not configurable yet)
- Global dashboard service to retrieve dashboard configuration

### CODE REVIEW

File: src/app/shared/interfaces.ts [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+  exp: number;
 }
 export interface IPaginationParams {
-    page: number;
-    pageSize: any;
+  page: number;
+  pageSize: any;
 }
 
 export interface Chart {
```
Comment (Sbrenthughes) [inline src/app/shared/interfaces.ts]: Can you rename this to IChart? Follows the interface pattern.

Comment (jhoelscher7) [reply src/app/shared/interfaces.ts]: I've seen both ways on this. https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names. If we want to stick with I prefixing them that is fine with me as well.

Comment (jhoelscher7) [reply src/app/shared/interfaces.ts]: Angular style basically says we can choose, so I will go ahead and switch it.
https://angular.io/guide/styleguide#style-03-03

Comment (Sbrenthughes) [reply src/app/shared/interfaces.ts]: Yeah one way or another we need to pick one so either adding the I or removing it for all of them :D

**Comment by Sbrenthughes** [comment]
I Approved but @nireeshT @kheinz04 can either of you run through it as well?

File: src/app/shared/charts/click-list/click-list.component.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 -->
 <div class="widget">
```
Comment (kheinz04) [inline src/app/shared/charts/click-list/click-list.component.html]: May need to revisit css classes for chart specific information

**Pull request merged**